src/unix/syscall.c

```diff
@@ -2509,14 +2507,9 @@ closure_function(0, 2, void, io_complete_ignore,
 static void syscall_schedule(context f, u64 call)
 {
     /* kernel context set on syscall entry */
-    if (kern_try_lock()) {
-        current_cpu()->state = cpu_kernel;
-        syscall_debug(f);
-    } else {
-        enqueue(runqueue, &current->deferred_syscall);
-        thread_pause(current);
-        runloop();
-    }
+    kern_lock();
```

Comment:
Yes, I found this to perform better for the webg test, and I thought it would apply in general to other syscall-intensive workloads where user threads are continuously suspended and resumed due to kernel-level events (such as network traffic). But yes, there may be other workloads where the try_lock() approach works better, I'm going to make this configurable.